packages/connector-chromium/src/connector.ts

```diff
@@ -0,0 +1,534 @@
+    }
+
+    private onError(error: Error) {
+        debug(`Error: ${error}`);
+    }
+
+    private async onRequest(request: puppeteer.Request) {
+        const requestUrl = request.url();
+        const event = { resource: requestUrl };
+
+        debug(`Request started: ${requestUrl}`);
+
+        if (request.isNavigationRequest()) {
+            this._headers = normalizeHeaders(request.headers())!;
+
+            await this._engine.emitAsync('fetch::start::target', event);
+        } else {
+            await this._engine.emitAsync('fetch::start', event);
+        }
+    }
+
+    private async onRequestFailed(request: puppeteer.Request) {
+        const response = request.response();
+        const resource = request.url();
+
+        if (response && response.status() >= 400) {
+            // Probably a 404, 503, and such, will be handled via fetch::end
+            return;
+        }
+
+        if (!this._dom) {
+            this._pendingRequests.push(this.onRequestFailed.bind(this, request));
+
+            return;
+        }
+
+        debug(`Request failed: ${resource}`);
+
+        const element: HTMLElement | null = this.getElementFromResponse(request);
+        const eventName = 'fetch::error';
+        const hops: string[] = request.redirectChain()
+            .map((redirect) => {
+                return redirect.url();
+            });
+
+        const event: FetchError = {
+            element,
+            error: request.failure(),
+            hops,
+            resource
+        };
+
+        await this._engine.emitAsync(eventName, event);
+    }
+
+    private async onResponse(response: puppeteer.Response) {
+        const resource = response.url();
+        const isTarget = response.request().isNavigationRequest();
+        const status = response.status();
+
+        debug(`Response received: ${resource}`);
+
+
+        if (status >= 300 && status < 400) {
+            // It's a redirect so it will be handled later
+            return;
+        }
+
+        if (isTarget) {
+            await this.navigationEnded(response);
+        } else if (!this._dom) {
+            // DOM isn't loaded yet, need to queue up
+            this._pendingRequests.push(this.onResponse.bind(this, response));
+
+            return;
+        }
+
+        // TODO: Check if it needs to be queue'd up and continue if not
+
+        const fetchEndPayload = await this.createFetchEndPayload(response);
+        /*
+         * If the target has a weird value like `application/x-httpd-php`
+         * (which translates into `unknown`) or is detected as `xml`.
+         * (e.g.: because it starts with
+         * `<?xml version="1.0" encoding="utf-8"?>` even though it has
+         * `<!DOCTYPE html>` declared after),
+         * we change the suffix to `html` so hints work properly.
+         */
+        let suffix = getType(fetchEndPayload.response.mediaType);
+        const defaults = ['unknown', 'xml'];
```

Comment:
We should fix this handling at the `getType` level so it's consistent across all connectors...

Reply:
OK, comming from `connector-chrome` but it makes sense.

Reply:
Not sure we can fix that TBH. This applies only to the target and `getType` doesn't have any knowledge of that. Transforming all `unknown` and/or `xml` to `html` doesn't seem like a good idea...